components/scream/src/mct_coupling/scream_scorpio_interface.F90

```diff
@@ -601,23 +654,27 @@ subroutine get_var(pio_file,varname,var)
   end subroutine get_var
 !=====================================================================!
   ! Lookup pointer for pio file based on filename.
-  subroutine get_pio_atm_file(filename,pio_atm_file,found)
+  subroutine get_pio_atm_file(filename,pio_file,found,purpose)
 
-    character(len=*),intent(in)   :: filename     ! Name of file to be found
-    type(pio_atm_output), pointer :: pio_atm_file ! Pointer to pio_atm_output structure associated with this filename
-    logical,optional,intent(out)  :: found        ! return TRUE if pio file already exists
+    character(len=*),intent(in)  :: filename     ! Name of file to be found
+    type(pio_atm_file), pointer  :: pio_file     ! Pointer to pio_atm_output structure associated with this filename
+    logical,optional,intent(out) :: found        ! return TRUE if pio file already exists
+    integer,optional,intent(in)  :: purpose      ! Purpose for this file lookup, 0 = as output, 1 = as input
```

Comment:
Since these interfaces are likely going to be used only inside scream infrastructure/AD stuff, let's choose safety over ease: let's make purpose a _mandatory_ argument, so we avoid "assuming" any intent.

Reply:
That's a good point.  I can make that change.